org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/ContentModelFormatterParticipant.java

```diff
@@ -67,4 +67,23 @@ public FormatElementCategory getFormatElementCategory(DOMElement element,
 		return null;
 	}
 
+	@Override
+	public boolean shouldCollapseEmptyElement(DOMElement element, SharedSettings sharedSettings) {
+		boolean enabled = sharedSettings.getFormattingSettings().isGrammarAwareFormatting();
+		if (!enabled) {
+			return false;

```

Comment:
I think it would be a good idea to collapse empty elements if the grammar aware formatting is off and emptyElements=collapse. My reasoning is that by default we have emptyElements=ignore and grammar aware formatting on, so if a user sets emptyElements=collapse and disables the grammar aware formatting, it's because they want to collapse empty elements even if it doesn't make sense with the grammar.

Reply:
What do you think about this Jessica, @angelozerr ?

Reply:
Right I agree, that makes sense

Reply:
I agree with you @datho7561.

Reply:
I think the collapse should be enabled for the most usecases. I have the impression is that the only usecase is when XML defines xsi:nil="true" and the XSD defines nillable="false", collapse should be disabled. I see just this usecase.